helix-gateway/src/main/java/org/apache/helix/gateway/util/StateTransitionMessageTranslateUtil.java

```diff
@@ -11,7 +15,24 @@ public static TransitionMessage translateSTMsgToProto() {
     return null;
   }
 
-  public static GatewayServiceManager.GateWayServiceEvent translateProtoToSTMsg(ShardStateMessage message) {
-    return null;
+  public static GatewayServiceEvent translateProtoToEvent(ShardStateMessage request) {
```

Comment:
I don't get this util at all...

If you would like to differentiate the event type, you can just make a full util and selectively pass in null value for the arguments are not necessary.